src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,9 +279,16 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
```

Comment:
instead of having this at multiple places, just have it once at the top of `GetIndexOfFirstNonAsciiByte_Sse2`. Similarly for `GetIndexOfFirstNonAsciiByte_Arm64`.
```csharp
if (!Sse2.IsSupported)
{
 throw new PlatformNotSupportedException();
}
```